modules/slugs/admin/slugs-admin.php

```diff
@@ -269,53 +263,55 @@ function qts_save_postdata( $post_id, $post = null ) {
     }
     $post_type_object = get_post_type_object( $post->post_type );
 
-    if ( ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE )                       // check autosave
-         || ( ! isset( $_POST['post_ID'] ) || $post_id != $_POST['post_ID'] ) // check revision
-         || ( isset( $_POST['qts_nonce'] ) && ! wp_verify_nonce( $_POST['qts_nonce'], 'qts_nonce' ) )   // verify nonce
-         || ( ! current_user_can( $post_type_object->cap->edit_post, $post_id ) ) ) {  // check permission
+    if ( ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE )
+         || ( ! isset( $_POST['post_ID'] ) || $post_id != $_POST['post_ID'] )
+         || ( isset( $_POST['qts_nonce'] ) && ! wp_verify_nonce( $_POST['qts_nonce'], 'qts_nonce' ) )
+         || ( ! current_user_can( $post_type_object->cap->edit_post, $post_id ) ) ) {
         return;
     }
     foreach ( $q_config['enabled_languages'] as $lang ) {
-
         // check required because it is not available inside quick edit
         if ( isset( $_POST["qts_{$lang}_slug"] ) ) {
-            $meta_value = apply_filters( 'qts_validate_post_slug', $_POST["qts_{$lang}_slug"], $post, $lang );
+            $slug = $_POST["qts_{$lang}_slug"];
+            $slug = qts_sanitize_post_slug( $slug, $post, $lang );
+            $slug = qts_unique_post_slug( $slug, $post, $lang );
+
             delete_post_meta( $post_id, QTS_META_PREFIX . $lang );
-            update_post_meta( $post_id, QTS_META_PREFIX . $lang, $meta_value );
+            update_post_meta( $post_id, QTS_META_PREFIX . $lang, $slug );
         }
     }
 }
 
 /**
- * Sanitize title as slug, if empty slug.
+ * Sanitize a term slug.
  *
- * @param $term (object) the term object
- * @param $slug (string) the slug name
- * @param $lang (string) the language
+ * @param string $slug the slug name
+ * @param WP_Term $term the term object
+ * @param string $lang the language
  *
- * @return string the slug validated
+ * @return string sanitized slug
  */
-function qts_validate_term_slug( $slug, $term, $lang ) {
+function qts_sanitize_term_slug( $slug, $term, $lang ) {
     global $q_config;
 
     $term_name = trim( qtranxf_use( $lang, $term->name, false, true ) );
-    if ( $term_name === '' ) {
+    if ( empty( $term_name ) ) {
         $term_name = trim( qtranxf_use( $q_config['default_language'], $term->name ) );
     }
     $slug = trim( $slug );
-    $slug = $slug === '' ? sanitize_title( $term_name ) : sanitize_title( $slug );
+    $slug = empty( $slug ) ? sanitize_title( $term_name ) : sanitize_title( $slug );
```

Comment:
This is not covering values of `$slug` that would be evaluated as `false`, like `'0'`. Better the previous test.

Reply:
Reverted.